# Investment Case Study
---

In [1]:
import numpy as np
import pandas as pd

# Checkpoint 1: Data Cleaning 1

In [2]:
# Creating the companies dataframe
companies = pd.read_csv('\\companies.txt',sep="\t",encoding='ISO-8859-1')


In [3]:
companies.isnull().sum()

permalink            0
name                 1
homepage_url      5058
category_list     3148
status               0
country_code      6958
state_code        8547
region            8030
city              8028
founded_at       15221
dtype: int64

In [4]:
companies.head(10)

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   
5     /Organization/01Games-Technology      01Games Technology   
6  /Organization/0Ndine-Biomedical-Inc  Ondine Biomedical Inc.   
7                 /Organization/0Xdata                  H2O.ai   
8                      /Organization/1                One Inc.   
9            /Organization/1-2-3-Listo             1,2,3 Listo   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   
5             http://www.01games.hk/   
6               http://ondinebio.com   
7                     http://h2o.ai/   
8                 http://whatis1.com   
9            http://www.123listo.com   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   
5                                              Games  operating          HKG   
6                                      Biotechnology  operating          CAN   
7                                          Analytics  operating          USA   
8                                             Mobile  operating          USA   
9                                         E-Commerce  operating          CHL   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         NaN  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         NaN  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010  
5        NaN              Hong Kong      Hong Kong         NaN  
6         BC              Vancouver      Vancouver  01-01-1997  
7         CA            SF Bay Area  Mountain View  01-01-2011  
8         CA            SF Bay Area  San Francisco  01-08-2011  
9         12               Santiago     Las Condes  01-01-2012

In [5]:
companies['permalink']=companies[['permalink']].applymap(lambda x: x.encode('utf-8').decode('ascii', 'ignore'))
companies['name']=companies.name.str.encode('utf-8').str.decode('ascii', 'ignore')

In [6]:
# Creating rounds2 dataframe
rounds2 = pd.read_csv('\\rounds2.csv',sep=",",encoding='ISO-8859-1')

In [7]:
rounds2.isnull().sum()

company_permalink              0
funding_round_permalink        0
funding_round_type             0
funding_round_code         83809
funded_at                      0
raised_amount_usd          19990
dtype: int64

In [8]:
rounds2.head(10)

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   
5       /ORGANIZATION/004-TECHNOLOGIES   
6     /organization/01games-technology   
7  /ORGANIZATION/0NDINE-BIOMEDICAL-INC   
8  /organization/0ndine-biomedical-inc   
9                 /ORGANIZATION/0XDATA   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9  /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   

  funding_round_code   funded_at  raised_amount_usd  
0                  B  05-01-2015         10000000.0  
1                  A  14-10-2014                NaN  
2                NaN  01-03-2014           700000.0  
3                  B  30-01-2014          3406878.0  
4                  A  19-03-2008          2000000.0  
5                NaN  24-07-2014                NaN  
6                NaN  01-07-2014            41250.0  
7                NaN  11-09-2009            43360.0  
8                NaN  21-12-2009           719491.0  
9                NaN  22-05-2013          3000000.0

In [9]:
# making modifiction for syncing company_permalink and funding_round_permalink column

rounds2['company_permalink']=rounds2[['company_permalink']].applymap(lambda x: x.encode('utf-8').decode('ascii', 'ignore'))
rounds2['funding_round_permalink']=rounds2[['funding_round_permalink']].applymap(lambda x: x.encode('utf-8').decode('ascii', 'ignore'))

In [10]:
rounds2.company_permalink = rounds2.company_permalink.apply(lambda x: x.lower())

In [11]:
companies.permalink = companies.permalink.apply(lambda x: x.lower())
companies['name']=companies['name'].str.lower()

In [12]:
len(rounds2.company_permalink.unique())

66368

1. How many unique companies are present in rounds2? = 66368

In [13]:
len(companies.name.unique())

66038

2. How many unique companies are present in companies file? = 66038

In [14]:
companies.loc[companies['permalink']=='/organization/tell-it-in',['name']]=' '.join(companies[companies['permalink']=='/organization/tell-it-in']['permalink'].str.split('/').tolist()[0][2].split('-'))

3. In the companies data frame, which column can be used as the  unique key for each company?
              Write the name of the column? = permalink

In [15]:
companies.loc[~rounds2.company_permalink.isin(companies.permalink)]

Empty DataFrame
Columns: [permalink, name, homepage_url, category_list, status, country_code, state_code, region, city, founded_at]
Index: []

4. Are there any companies in the rounds2 file which are not present in companies?
    Answer Y/N = N

In [16]:
master_frame = pd.merge(rounds2,companies, left_on="company_permalink", right_on="permalink", how='left')

In [17]:
master_frame.shape

(114949, 16)

5. Question. Merge the two data frames so that all variables (columns) in the companies frame are added to the rounds2 data frame.
    Name the merged frame master_frame.
    How many observations are present in master_frame ?=
    There are 114949 rows, 16 columns.
   

In [18]:
master_frame.isnull().sum()

company_permalink              0
funding_round_permalink        0
funding_round_type             0
funding_round_code         83809
funded_at                      0
raised_amount_usd          19990
permalink                      0
name                           0
homepage_url                6134
category_list               3410
status                         0
country_code                8678
state_code                 10946
region                     10167
city                       10164
founded_at                 20521
dtype: int64

In [19]:
master_frame = master_frame[master_frame['status']!='closed']

In [20]:
master_frame[master_frame['raised_amount_usd'].isna()]

company_permalink  \
1                                  /organization/-qounter   
5                          /organization/004-technologies   
14                                        /organization/1   
17                                  /organization/1-4-all   
18                         /organization/1-618-technology   
19                            /organization/1-800-dentist   
33                                /organization/1000lookz   
40                          /organization/1000museums-com   
42                          /organization/1000museums-com   
51                                  /organization/1006-tv   
53                                  /organization/1006-tv   
54                                 /organization/100du-tv   
63                                  /organization/10alike   
64                                  /organization/10alike   
67                                    /organization/10sec   
70                                    /organization/10six   
71                               /organization/10tracks-2   
72                               /organization/10tracks-2   
77                                  /organization/10north   
79                        /organization/115-network-disks   
83                         /organization/12-star-survival   
84                               /organization/120-sports   
85                              /organization/121-rentals   
98                           /organization/123contactform   
101                               /organization/123people   
107                               /organization/12society   
117                                 /organization/139shop   
122                                    /organization/1417   
123                                   /organization/15fen   
128                              /organization/15shopstop   
...                                                   ...   
114765                                 /organization/zuli   
114774                           /organization/zumata-ltd   
114776                        /organization/zumba-fitness   
114793                            /organization/zumodrive   
114806                            /organization/zuoyebang   
114807                    /organization/zup-it-innovation   
114809                               /organization/zupcat   
114815                                 /organization/zura   
114818                         /organization/zurex-pharma   
114820                                /organization/zurff   
114822  /organization/zurn-international-e-commerce-co...   
114823                         /organization/zurrba-group   
114827                                 /organization/zuse   
114833                                /organization/zuuka   
114834                                /organization/zuuka   
114847                                /organization/zvooq   
114857                               /organization/zyante   
114859                                  /organization/zyb   
114860                                /organization/zyfin   
114878                       /organization/zyme-solutions   
114884                             /organization/zymergen   
114895                             /organization/zymochem   
114896                                /organization/zympi   
114897                            /organization/zymr-inc-   
114898                            /organization/zymr-inc-   
114899                            /organization/zymr-inc-   
114905            /organization/zynergy-projects-services   
114913                                /organization/zynga   
114927                               /organization/zypsee   
114946                                 /organization/eron   

                                funding_round_permalink   funding_round_type  \
1       /funding-round/22dacff496eb7acb2b901dec1dfe5633              venture   
5       /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830              venture   
14      /funding-

In [21]:
master_frame.shape

(106224, 16)

In [22]:
master_frame[master_frame.status == 'closed']

Empty DataFrame
Columns: [company_permalink, funding_round_permalink, funding_round_type, funding_round_code, funded_at, raised_amount_usd, permalink, name, homepage_url, category_list, status, country_code, state_code, region, city, founded_at]
Index: []

In [23]:
master_frame = master_frame[pd.notnull(master_frame['raised_amount_usd'])]

In [24]:
master_frame.shape

(87903, 16)

In [25]:
round((master_frame.isnull().sum()/len(master_frame.index))*100,2)

company_permalink           0.00
funding_round_permalink     0.00
funding_round_type          0.00
funding_round_code         70.21
funded_at                   0.00
raised_amount_usd           0.00
permalink                   0.00
name                        0.00
homepage_url                4.31
category_list               0.72
status                      0.00
country_code                5.26
state_code                  7.10
region                      6.50
city                        6.50
founded_at                 15.82
dtype: float64

In [26]:
len(master_frame.name.unique())

48814

In [27]:
master_frame['raised_amount_usd']=master_frame['raised_amount_usd'].apply(lambda val: round(val/1000000,3))

In [28]:
master_frame=master_frame.rename(columns = {'raised_amount_usd':'raised_amount_mil_usd'})

In [29]:
master_frame

company_permalink  \
0                                     /organization/-fame   
2                                  /organization/-qounter   
3                     /organization/-the-one-of-them-inc-   
4                                   /organization/0-6-com   
6                        /organization/01games-technology   
7                     /organization/0ndine-biomedical-inc   
8                     /organization/0ndine-biomedical-inc   
9                                    /organization/0xdata   
10                                   /organization/0xdata   
11                                   /organization/0xdata   
12                                   /organization/0xdata   
13                                        /organization/1   
15                                        /organization/1   
16                              /organization/1-2-3-listo   
20                            /organization/1-800-doctors   
21               /organization/1-800-publicrelations-inc-   
22                             /organization/1-mainstream   
23                                  /organization/1-of-99   
24                              /organization/10-20-media   
25                              /organization/10-20-media   
26                              /organization/10-20-media   
27                              /organization/10-20-media   
28                          /organization/10-minutes-with   
29                          /organization/10-minutes-with   
30                               /organization/1000-corks   
31                             /organization/1000-markets   
32                                  /organization/1000chi   
34                             /organization/1000memories   
35                             /organization/1000memories   
36                          /organization/1000museums-com   
...                                                   ...   
114912                                /organization/zynga   
114914                                /organization/zynga   
114915                             /organization/zyngenia   
114916                              /organization/zynstra   
114917                              /organization/zynstra   
114918                              /organization/zynstra   
114919                              /organization/zynstra   
114920                           /organization/zyomyx-inc   
114921                           /organization/zyomyx-inc   
114922                           /organization/zyomyx-inc   
114923                           /organization/zyomyx-inc   
114924                                 /organization/zype   
114925                                 /organization/zype   
114926                               /organization/zypsee   
114932                                /organization/zyrra   
114933                                /organization/zyrra   
114934                                /organization/zyrra   
114935                                /organization/zyrra   
114936                               /organization/zystor   
114937                           /organization/zytoprotec   
114938                                /organization/zzish   
114939                                /organization/zzish   
114940                                /organization/zzish   
114941  /organization/zznode-science-and-technology-co...   
114942                          /organization/zzzzapp-com   
114943                          /organization/zzzzapp-com   
114944                          /organization/zzzzapp-com   
114945                          /organization/zzzzapp-com   
114947                               /organization/asys-2   
114948  /organization/novatiff-reklam-ve-tantm-hizmetl...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2       /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3       /funding-round/

# Checkpoint 2: Funding Type Analysis 

In [30]:
master_frame.groupby('funding_round_type')['raised_amount_mil_usd'].mean()

funding_round_type
angel                      0.979917
convertible_note           1.479851
debt_financing            17.774072
equity_crowdfunding        0.519852
grant                      4.154310
non_equity_assistance      0.402853
post_ipo_debt            174.609345
post_ipo_equity           84.233854
private_equity            75.575805
product_crowdfunding       1.411511
secondary_market          91.439808
seed                       0.738591
undisclosed               18.837079
venture                   11.895604
Name: raised_amount_mil_usd, dtype: float64

In [31]:
master_frame

company_permalink  \
0                                     /organization/-fame   
2                                  /organization/-qounter   
3                     /organization/-the-one-of-them-inc-   
4                                   /organization/0-6-com   
6                        /organization/01games-technology   
7                     /organization/0ndine-biomedical-inc   
8                     /organization/0ndine-biomedical-inc   
9                                    /organization/0xdata   
10                                   /organization/0xdata   
11                                   /organization/0xdata   
12                                   /organization/0xdata   
13                                        /organization/1   
15                                        /organization/1   
16                              /organization/1-2-3-listo   
20                            /organization/1-800-doctors   
21               /organization/1-800-publicrelations-inc-   
22                             /organization/1-mainstream   
23                                  /organization/1-of-99   
24                              /organization/10-20-media   
25                              /organization/10-20-media   
26                              /organization/10-20-media   
27                              /organization/10-20-media   
28                          /organization/10-minutes-with   
29                          /organization/10-minutes-with   
30                               /organization/1000-corks   
31                             /organization/1000-markets   
32                                  /organization/1000chi   
34                             /organization/1000memories   
35                             /organization/1000memories   
36                          /organization/1000museums-com   
...                                                   ...   
114912                                /organization/zynga   
114914                                /organization/zynga   
114915                             /organization/zyngenia   
114916                              /organization/zynstra   
114917                              /organization/zynstra   
114918                              /organization/zynstra   
114919                              /organization/zynstra   
114920                           /organization/zyomyx-inc   
114921                           /organization/zyomyx-inc   
114922                           /organization/zyomyx-inc   
114923                           /organization/zyomyx-inc   
114924                                 /organization/zype   
114925                                 /organization/zype   
114926                               /organization/zypsee   
114932                                /organization/zyrra   
114933                                /organization/zyrra   
114934                                /organization/zyrra   
114935                                /organization/zyrra   
114936                               /organization/zystor   
114937                           /organization/zytoprotec   
114938                                /organization/zzish   
114939                                /organization/zzish   
114940                                /organization/zzish   
114941  /organization/zznode-science-and-technology-co...   
114942                          /organization/zzzzapp-com   
114943                          /organization/zzzzapp-com   
114944                          /organization/zzzzapp-com   
114945                          /organization/zzzzapp-com   
114947                               /organization/asys-2   
114948  /organization/novatiff-reklam-ve-tantm-hizmetl...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2       /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3       /funding-round/

In [32]:
master_frame[master_frame['country_code'].isnull()]

company_permalink  \
3                     /organization/-the-one-of-them-inc-   
32                                  /organization/1000chi   
66                                    /organization/10k-2   
95                                /organization/1234enter   
96                                /organization/1234enter   
100                            /organization/123greetings   
127                            /organization/15minutesnow   
133                                /organization/17-media   
145                             /organization/1800diapers   
173                                    /organization/1lay   
174                                    /organization/1lay   
186                                    /organization/1sdk   
187                                    /organization/1sdk   
205                       /organization/2-pro-media-group   
386                                 /organization/360quan   
387                                 /organization/360quan   
388                                 /organization/360quan   
447     /organization/3d-intelligent-pathological-medi...   
483                                 /organization/3dmgame   
489                               /organization/3dprintus   
496                                 /organization/3dvista   
516                                     /organization/3lm   
571                           /organization/4-antibody-ag   
581                               /organization/404-found   
582                               /organization/404-found   
641                                   /organization/4less   
649                                   /organization/4plat   
666                               /organization/4thetable   
674                   /organization/5-cups-and-some-sugar   
694                               /organization/500indies   
...                                                   ...   
113967                                 /organization/zepp   
113968                                 /organization/zepp   
113978                            /organization/zerg--inc   
114009                            /organization/zerobound   
114116                        /organization/zevan-limited   
114134       /organization/zhejiang-xianju-pharmaceutical   
114155                              /organization/zhongan   
114161                       /organization/zhongyou-group   
114183                               /organization/ziebel   
114206                           /organization/ziggli-inc   
114207                           /organization/ziggli-inc   
114218        /organization/zigya-technology-labs-pvt-ltd   
114265                               /organization/zimory   
114266                               /organization/zimory   
114301                            /organization/zingbox-2   
114302                          /organization/zingbox-ltd   
114377                             /organization/zipmatch   
114378                             /organization/zipmatch   
114394                 /organization/zippy-com-au-pty-ltd   
114470                         /organization/zmqnw-com-cn   
114512                               /organization/zolk-c   
114530                                  /organization/zon   
114551                                /organization/zoobe   
114569                              /organization/zoom-tv   
114646                             /organization/zootcard   
114697                              /organization/zoweetv   
114736                     /organization/zte9-corporation   
114752                                /organization/zuggi   
114849      /organization/zwayo-"on-demand-valet-parking"   
114948  /organization/novatiff-reklam-ve-tantm-hizmetl...   

                                funding_round_permalink funding_round_type  \
3       /funding-round/650b8f704416801069bb178a1418776b            venture   
32      /funding-round/854fae13c5b26b2ea6b32b06c2e5890c              angel   
66      /funding-round/

In [33]:
master_frame.category_list.isna().count()

87903

In [34]:
master_frame.shape

(87903, 16)

In [35]:
master_frame = master_frame[pd.notnull(master_frame['country_code'])]

In [36]:
master_frame.shape

(83277, 16)

In [37]:
len(master_frame.name.unique())

45289

In [38]:
master_frame.groupby('funding_round_type')['raised_amount_mil_usd'].mean()

funding_round_type
angel                      0.994756
convertible_note           1.340013
debt_financing            17.891010
equity_crowdfunding        0.511266
grant                      4.286176
non_equity_assistance      0.460610
post_ipo_debt            175.469739
post_ipo_equity           67.093605
private_equity            75.797196
product_crowdfunding       1.521047
secondary_market          94.089400
seed                       0.763097
undisclosed               16.018211
venture                   11.903153
Name: raised_amount_mil_usd, dtype: float64

In [39]:
master_frame_vent = master_frame[master_frame.funding_round_type == 'venture']

# Checkpoint 3: Country Analysis

In [40]:
master_frame_vent.isna().sum()

company_permalink              0
funding_round_permalink        0
funding_round_type             0
funding_round_code         19874
funded_at                      0
raised_amount_mil_usd          0
permalink                      0
name                           0
homepage_url                1983
category_list                201
status                         0
country_code                   0
state_code                   696
region                       549
city                         549
founded_at                  7506
dtype: int64

In [41]:
 top9 = master_frame_vent.groupby('country_code',as_index=False)['raised_amount_mil_usd'].sum().sort_values(by = ['raised_amount_mil_usd'],ascending=False).head(9)
top9['Rank'] = range(1,10)
top9.set_index(['Rank'],inplace = True)

In [42]:
master_frame_vent

company_permalink  \
0                                     /organization/-fame   
4                                   /organization/0-6-com   
8                     /organization/0ndine-biomedical-inc   
10                                   /organization/0xdata   
11                                   /organization/0xdata   
12                                   /organization/0xdata   
22                             /organization/1-mainstream   
28                          /organization/10-minutes-with   
34                             /organization/1000memories   
38                          /organization/1000museums-com   
39                          /organization/1000museums-com   
41                          /organization/1000museums-com   
44                          /organization/1000museums-com   
46                               /organization/1001-menus   
47                               /organization/1001-menus   
49                               /organization/1001-menus   
50                       /organization/1001pharmacies-com   
52                                  /organization/1006-tv   
55                                 /organization/100du-tv   
56                                 /organization/100e-com   
57                                 /organization/100e-com   
59                                /organization/100health   
61                                  /organization/100plus   
62                                 /organization/1010data   
73                             /organization/10x-genomics   
74                         /organization/10x-technologies   
80                                    /organization/117go   
97                           /organization/123contactform   
99                              /organization/123feng-com   
103                          /organization/128-technology   
...                                                   ...   
114881                       /organization/zyme-solutions   
114883                             /organization/zymergen   
114889                            /organization/zymeworks   
114890                            /organization/zymeworks   
114891                            /organization/zymeworks   
114892                            /organization/zymeworks   
114893                            /organization/zymeworks   
114894                            /organization/zymeworks   
114900                                /organization/zyncd   
114902                               /organization/zyncro   
114903                               /organization/zyncro   
114904              /organization/zynerba-pharmaceuticals   
114906                                /organization/zynga   
114907                                /organization/zynga   
114908                                /organization/zynga   
114909                                /organization/zynga   
114910                                /organization/zynga   
114911                                /organization/zynga   
114912                                /organization/zynga   
114914                                /organization/zynga   
114915                             /organization/zyngenia   
114916                              /organization/zynstra   
114917                              /organization/zynstra   
114920                           /organization/zyomyx-inc   
114922                           /organization/zyomyx-inc   
114923                           /organization/zyomyx-inc   
114934                                /organization/zyrra   
114936                               /organization/zystor   
114937                           /organization/zytoprotec   
114941  /organization/zznode-science-and-technology-co...   

                                funding_round_permalink funding_round_type  \
0       /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
4       /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
8       /funding-round/

In [43]:
top9

country_code  raised_amount_mil_usd
Rank                                    
1             USA             399935.179
2             CHN              39101.773
3             GBR              18241.345
4             IND              14050.604
5             CAN               8825.477
6             FRA               6775.039
7             ISR               6503.020
8             DEU               6061.141
9             JPN               3242.031

#### Answers to the questions:
    1. Top English speaking country = USA
    2. Second English speaking country = GBR
    3. Third English speaking country = IND 

In [44]:
len(top9.country_code.unique())

9

In [45]:
master_frame.shape

(83277, 16)

# Checkpoint 4: Sector Analysis 1

In [46]:
mapping=pd.read_csv('\\mapping.csv')

In [47]:
mapping

category_list  Automotive & Sports  Blanks  \
0                           NaN                    0       1   
1                            3D                    0       0   
2                   3D Printing                    0       0   
3                 3D Technology                    0       0   
4                    Accounting                    0       0   
5              Active Lifestyle                    0       0   
6                  Ad Targeting                    0       0   
7            Advanced Materials                    0       0   
8              Adventure Travel                    1       0   
9                   Advertising                    0       0   
10        Advertising Exchanges                    0       0   
11         Advertising Networks                    0       0   
12        Advertising Platforms                    0       0   
13                       Advice                    0       0   
14                    Aerospace                    1       0   
15                  Agriculture                    0       0   
16        Air Pollution Control                    0       0   
17                   Algorithms                    0       0   
18                  All Markets                    0       0   
19                 All Students                    0       0   
20          Alter0tive Medicine                    0       0   
21                       Alumni                    0       0   
22                     A0lytics                    0       0   
23                      Android                    0       0   
24                       Angels                    0       0   
25                  Animal Feed                    0       0   
26   Anything Capital Intensive                    0       0   
27                App Discovery                    0       0   
28                App Marketing                    0       0   
29                   App Stores                    0       0   
..                          ...                  ...     ...   
658                   Usability                    0       0   
659      User Experience Design                    0       0   
660              User Interface                    0       0   
661                   Utilities                    0       0   
662     Vending and Concessions                    0       0   
663             Venture Capital                    0       0   
664                   Veteri0ry                    0       0   
665                       Video                    0       0   
666          Video Conferencing                    0       0   
667                 Video Games                    0       0   
668             Video on Demand                    0       0   
669             Video Streaming                    0       0   
670          Virtual Workforces                    0       0   
671                        VoIP                    0       0   
672             Waste Ma0gement                    0       0   
673                       Watch                    0       0   
674                       Water                    0       0   
675          Water Purification                    0       0   
676                   Wearables                    0       0   
677                Web Browsers                    0       0   
678                  Web Design                    0       0   
679             Web Development                    0       0   
680                 Web Hosting                    0       0   
681                   Web Tools                    0       0   
682                    Weddings                    0       0   
683                   Wholesale                    0       0   
684            Wine And Spirits                    0       0   
685                    Wireless                    0       0   
686                       Women                    0       0   
687                Young Adults                    0       0   

     Cleantech / Semiconductors  Entertainment  H

In [48]:
mapping.isna().sum()

category_list                              1
Automotive & Sports                        0
Blanks                                     0
Cleantech / Semiconductors                 0
Entertainment                              0
Health                                     0
Manufacturing                              0
News, Search and Messaging                 0
Others                                     0
Social, Finance, Analytics, Advertising    0
dtype: int64

In [49]:
prim_sector = master_frame_vent['category_list'].str.split('|',n=1,expand=True)

In [50]:
master_frame_vent.shape

(44949, 16)

In [51]:
prim_sector.shape

(44949, 2)

In [52]:
type(prim_sector)

pandas.core.frame.DataFrame

In [53]:
master_frame_vent['Primary_Sector']=prim_sector[0]

D:\Visual Studio\Shared\Anaconda3_64\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [54]:
mapping['category_list']=mapping['category_list'].str.replace('0','na')

In [55]:
mapping.columns

Index(['category_list', 'Automotive & Sports', 'Blanks',
       'Cleantech / Semiconductors', 'Entertainment', 'Health',
       'Manufacturing', 'News, Search and Messaging', 'Others',
       'Social, Finance, Analytics, Advertising'],
      dtype='object')

In [56]:
melted=pd.melt(mapping, id_vars=['category_list'],value_vars=['Automotive & Sports', 'Blanks',
       'Cleantech / Semiconductors', 'Entertainment', 'Health',
       'Manufacturing', 'News, Search and Messaging', 'Others',
       'Social, Finance, Analytics, Advertising'])

In [57]:
melted.head(30)

category_list             variable  value
0                          NaN  Automotive & Sports      0
1                           3D  Automotive & Sports      0
2                  3D Printing  Automotive & Sports      0
3                3D Technology  Automotive & Sports      0
4                   Accounting  Automotive & Sports      0
5             Active Lifestyle  Automotive & Sports      0
6                 Ad Targeting  Automotive & Sports      0
7           Advanced Materials  Automotive & Sports      0
8             Adventure Travel  Automotive & Sports      1
9                  Advertising  Automotive & Sports      0
10       Advertising Exchanges  Automotive & Sports      0
11        Advertising Networks  Automotive & Sports      0
12       Advertising Platforms  Automotive & Sports      0
13                      Advice  Automotive & Sports      0
14                   Aerospace  Automotive & Sports      1
15                 Agriculture  Automotive & Sports      0
16       Air Pollution Control  Automotive & Sports      0
17                  Algorithms  Automotive & Sports      0
18                 All Markets  Automotive & Sports      0
19                All Students  Automotive & Sports      0
20        Alternative Medicine  Automotive & Sports      0
21                      Alumni  Automotive & Sports      0
22                   Analytics  Automotive & Sports      0
23                     Android  Automotive & Sports      0
24                      Angels  Automotive & Sports      0
25                 Animal Feed  Automotive & Sports      0
26  Anything Capital Intensive  Automotive & Sports      0
27               App Discovery  Automotive & Sports      0
28               App Marketing  Automotive & Sports      0
29                  App Stores  Automotive & Sports      0

In [58]:
melted=melted[melted.value==1]

In [59]:
melted=melted.rename(columns={'category_list':'Primary_Sector','variable':'main_sector'})

In [60]:
melted=melted.drop('value',axis=1)

In [61]:
master_FT = pd.merge(master_frame_vent,melted,on='Primary_Sector',how='inner')

In [62]:
master_FT.groupby('main_sector').main_sector.count()

main_sector
Automotive & Sports                          690
Blanks                                       201
Cleantech / Semiconductors                  9274
Entertainment                               2521
Health                                      3742
Manufacturing                               3075
News, Search and Messaging                  5483
Others                                     10680
Social, Finance, Analytics, Advertising     9107
Name: main_sector, dtype: int64

In [63]:
master_FT=master_FT[master_FT.main_sector!='Blanks']

In [64]:
master_FT.groupby('main_sector').main_sector.count()

main_sector
Automotive & Sports                          690
Cleantech / Semiconductors                  9274
Entertainment                               2521
Health                                      3742
Manufacturing                               3075
News, Search and Messaging                  5483
Others                                     10680
Social, Finance, Analytics, Advertising     9107
Name: main_sector, dtype: int64

In [65]:
master_FT.shape

(44572, 18)

In [66]:
master_frame_vent.shape

(44949, 17)

# Checkpoint 5: Sector Analysis 2

In [67]:
master_FT_5M_15M = master_FT[master_FT['raised_amount_mil_usd'].between(5.0,15.0,inclusive=True)]

In [68]:
master_FT_5M_15M.shape

(14756, 18)

In [69]:
D1 = master_FT_5M_15M[master_FT_5M_15M['country_code']=='USA']
D1 = D1.assign(count_of_invest =D1.groupby('main_sector')['main_sector'].transform('count'))
D1 = D1.assign(sum_of_invest =D1.groupby('main_sector')['raised_amount_mil_usd'].transform('sum'))
#D1

In [70]:
D2 = master_FT_5M_15M[master_FT_5M_15M['country_code']=='GBR']
D2 = D2.assign(count_of_invest =D2.groupby('main_sector')['main_sector'].transform('count'))
D2 = D2.assign(sum_of_invest =D2.groupby('main_sector')['raised_amount_mil_usd'].transform('sum'))
#D2

In [71]:
D3 = master_FT_5M_15M[master_FT_5M_15M['country_code']=='IND']
D3 = D3.assign(count_of_invest =D3.groupby('main_sector')['main_sector'].transform('count'))
D3 = D3.assign(sum_of_invest =D3.groupby('main_sector')['raised_amount_mil_usd'].transform('sum'))
#D3

In [72]:
master_FT_5M_15M[master_FT_5M_15M['main_sector'].isnull()]

Empty DataFrame
Columns: [company_permalink, funding_round_permalink, funding_round_type, funding_round_code, funded_at, raised_amount_mil_usd, permalink, name, homepage_url, category_list, status, country_code, state_code, region, city, founded_at, Primary_Sector, main_sector]
Index: []

In [73]:
D1

company_permalink  \
4               /organization/all-def-digital   
14                   /organization/chefs-feed   
20               /organization/huffingtonpost   
21               /organization/huffingtonpost   
30                    /organization/mediabong   
33              /organization/newscorporation   
34                      /organization/nokeena   
36                      /organization/nokeena   
47                        /organization/vadio   
77                       /organization/abilto   
78                       /organization/abilto   
80                        /organization/abine   
81                      /organization/aboutme   
82                      /organization/aboutme   
88                     /organization/aeropost   
90               /organization/akonix-systems   
91              /organization/allbusiness-com   
92              /organization/allbusiness-com   
94                   /organization/amberpoint   
95                   /organization/amberpoint   
104               /organization/apartmentlist   
107           /organization/associatedcontent   
108           /organization/associatedcontent   
109           /organization/associatedcontent   
110               /organization/assured-labor   
111               /organization/assured-labor   
112                      /organization/assurz   
126                     /organization/bambeco   
139                   /organization/beliefnet   
147               /organization/bigtentdesign   
...                                       ...   
44642             /organization/patientkeeper   
44643             /organization/patientkeeper   
44645             /organization/patientkeeper   
44646             /organization/patientkeeper   
44647             /organization/patientkeeper   
44648             /organization/patientkeeper   
44650                 /organization/perceptis   
44652  /organization/secureone-data-solutions   
44664       /organization/planetary-resources   
44670                 /organization/predictix   
44673                     /organization/prezi   
44677       /organization/sunverge-energy-inc   
44679                 /organization/responsys   
44696                     /organization/roost   
44697                     /organization/roost   
44698                     /organization/topix   
44700                    /organization/rxwiki   
44701       /organization/tikvah-therapeutics   
44703                 /organization/satmetrix   
44705                 /organization/satmetrix   
44711                 /organization/switchfly   
44713                 /organization/switchfly   
44719     /organization/schweiger-dermatology   
44745         /organization/tactus-technology   
44747                  /organization/thisnext   
44750                   /organization/thredhq   
44755    /organization/wizrocket-technologies   
44760            /organization/user-interface   
44762                    /organization/washio   
44766                     /organization/yapta   

                               funding_round_permalink funding_round_type  \
4      /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
14     /funding-round/adca195749ae9ace84684723fbe75e5b            venture   
20     /funding-round/7f05940c4d2dfecb8e50a0e5720e5065            venture   
21     /funding-round/9241ae16e08df17ebdc064e49e23035a            venture   
30     /funding-round/9282890ca87072025dc1807f400acee6            venture   
33     /funding-round/8f6d7c4592e43e91e8688ba342bffcb7            venture   
34     /funding-round/9225f2db6b1b74892d5de6a8744b94ea            venture   
36     /funding-round/fb7a617ed6d1a3203024f0f111417bfb            venture   
47     /funding-round/e1570c5060bf30b9972a2b0dc0562b02            venture   
77     /funding-round/3493aef8ac2dd0e24ebe66801dc28fde            venture   
78     /funding-round/7cf800ea20eaa3e9cae70dc2f862b13e            venture   
80     /funding-round/6dd104a50b0547bd870cea2ae41a76b1           

1. Total number of investments (count)

In [74]:
# Country 1
D1.shape[0]

11269

In [75]:
# Country 2
D2.shape[0]

582

In [76]:
# Country 3
D3.shape[0]

315

2. Total amount of investment (USD)

In [77]:
# Country 1
D1.raised_amount_mil_usd.sum()

100851.886

In [78]:
# Country 2
D2.raised_amount_mil_usd.sum()

5061.262000000001

In [79]:
# Country 3
D3.raised_amount_mil_usd.sum()

2820.674

3. Top sector (based on count of investments)

In [80]:
#Country 1
top_sector_D1=D1.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3).index[0]
top_sector_D1

'Others'

In [81]:
D1.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3)

main_sector
Others                                     2765
Social, Finance, Analytics, Advertising    2594
Cleantech / Semiconductors                 2152
Name: main_sector, dtype: int64

In [82]:
# Country 2
top_sector_D2=D2.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3).index[0]
top_sector_D2

'Others'

In [83]:
# Country 3
top_sector_D3=D3.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3).index[0]
top_sector_D3

'Others'

4. Second-best sector (based on count of investments)

In [84]:
#Country 1
sec_best_D1=D1.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3).index[1]
sec_best_D1

'Social, Finance, Analytics, Advertising'

In [85]:
# Country 2
sec_best_D2=D2.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3).index[1]
sec_best_D2

'Social, Finance, Analytics, Advertising'

In [86]:
# Country 3
sec_best_D3=D3.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3).index[1]
sec_best_D3

'Social, Finance, Analytics, Advertising'

5. Third-best sector (based on count of investments)

In [87]:
#Country 1
third_best_D1=D1.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3).index[2]
third_best_D1

'Cleantech / Semiconductors'

In [88]:
# Country 2
third_best_D2=D2.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3).index[2]
third_best_D2

'Cleantech / Semiconductors'

In [89]:
# Country 3
third_best_D3=D3.groupby('main_sector')['main_sector'].count().sort_values(ascending=False).nlargest(3).index[2]
third_best_D3

'News, Search and Messaging'

6. Number of investments in the top sector (refer to point 3)

In [90]:
# Country 1
D1[D1.main_sector == top_sector_D1].shape[0]

2765

In [91]:
# Country 2
D2[D2.main_sector == top_sector_D2].shape[0]

141

In [92]:
# Country 3
D3[D3.main_sector == top_sector_D3].shape[0]

103

7. Number of investments in the second-best sector (refer to point 4)

In [93]:
# Country 1
D1[D1.main_sector == sec_best_D1].shape[0]

2594

In [94]:
# Country 2
D2[D2.main_sector == sec_best_D2].shape[0]

128

In [95]:
# Country 3
D3[D3.main_sector == sec_best_D3].shape[0]

59

8. Number of investments in the third-best sector (refer to point 5)

In [96]:
# Country 1
D1[D1.main_sector == third_best_D1].shape[0]

2152

In [97]:
# Country 2
D2[D2.main_sector == third_best_D2].shape[0]

115

In [98]:
# Country 3
D3[D3.main_sector == third_best_D3].shape[0]

49

9. For the top sector count-wise (point 3), which company received the highest investment?

In [99]:
# Country 1
D1[D1.main_sector == top_sector_D1].sort_values(by = 'raised_amount_mil_usd', ascending = False).head(1)['name'].values[0]

'udacity'

In [100]:
# Country 2
D2[D2.main_sector == top_sector_D2].sort_values(by = 'raised_amount_mil_usd', ascending = False).head(1)['name'].values[0]

'tribold'

In [101]:
# Country 3

D3[D3.main_sector == top_sector_D3].sort_values(by = 'raised_amount_mil_usd', ascending = False).head(1)['name'].values[0]

'k-12 techno services'

10. For the second-best sector count-wise (point 4), which company received the highest investment?

In [102]:
# Country 1
D1[D1.main_sector == sec_best_D1].sort_values(by = 'raised_amount_mil_usd', ascending = False).head(1)['name'].values[0]

'livefyre'

In [103]:
# Country 2
D2[D2.main_sector == sec_best_D2].sort_values(by = 'raised_amount_mil_usd', ascending = False).head(1)['name'].values[0]

'similarweb'

In [104]:
# Country 3
D3[D3.main_sector == sec_best_D3].sort_values(by = 'raised_amount_mil_usd', ascending = False).head(1)['name'].values[0]

'apnapaisa'